components/script/dom/bluetoothdevice.rs

```diff
@@ -133,6 +150,41 @@ impl BluetoothDevice {
         descriptor_map.insert(descriptor.instance_id.clone(), MutHeap::new(&bt_descriptor));
         return bt_descriptor;
     }
+
+
+    // https://webbluetoothcg.github.io/web-bluetooth/#clean-up-the-disconnected-device
+    #[allow(unrooted_must_root)]
+    pub fn clean_up_disconnected_device(&self) {
+        // Step 1.
+        self.Gatt().set_connected(false);
+
+        // TODO: Step 2: Implement activeAlgorithms internal slot for BluetoothRemoteGATTServer.
+
+        // Step 3: We don't need `context`, we get the attributeInstanceMap from the device.
+        // https://github.com/WebBluetoothCG/web-bluetooth/issues/330
+
+        let mut service_map = self.attribute_instance_map.0.borrow_mut();
+
+        // Step 4, 5.
+        let _ = service_map.drain().map(|(_, s)| s.get().set_represented_service_to_null()).collect::<Vec<()>>();
```

Comment:
If we don't have to check if it succeeds, why not send 3 vec to the lower level like `SetRepresentedToNull(Vec<dev>, Vec<char>, Vec<desc>, sender)` then we can remove the `GATTLevel Enum`. (I don't see where you use the the Device enum)